src/Instrumentation/Laravel/tests/Integration/Queue/QueueTest.php

```diff
@@ -0,0 +1,137 @@
+<?php
+
+declare(strict_types=1);
+
+namespace OpenTelemetry\Tests\Contrib\Instrumentation\Laravel\Integration\Queue;
+
+use DateInterval;
+use DateTimeImmutable;
+use Illuminate\Contracts\Queue\Queue;
+use Illuminate\Queue\RedisQueue;
+use Illuminate\Queue\SqsQueue;
+use Illuminate\Redis\Connections\Connection;
+use Mockery\MockInterface;
+use OpenTelemetry\SDK\Common\Time\ClockFactory;
+use OpenTelemetry\SDK\Common\Time\ClockInterface;
+use OpenTelemetry\SemConv\TraceAttributes;
+use OpenTelemetry\Tests\Contrib\Instrumentation\Laravel\Fixtures\Jobs\DummyJob;
+use OpenTelemetry\Tests\Contrib\Instrumentation\Laravel\Integration\TestCase;
+use Psr\Log\LoggerInterface;
+
+class QueueTest extends TestCase
+{
+    private Queue $queue;
+
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        /** @psalm-suppress PossiblyNullReference */
+        $this->queue = $this->app->make(Queue::class);
+    }
+
+    /**
+     * @test
+     */
+    public function it_handles_pushing_to_a_queue(): void
+    {
+        $this->queue->push(new DummyJob('A'));
+        $this->queue->push(function (LoggerInterface $logger) {
+            $logger->info('Logged from closure');
+        });
+
+        $this->assertEquals('sync process', $this->storage[0]->getName());
+        $this->assertEquals('Task: A', $this->storage[0]->getEvents()[0]->getName());
+
+        $this->assertEquals('sync process', $this->storage[1]->getName());
+        $this->assertEquals('Logged from closure', $this->storage[1]->getEvents()[0]->getName());
+    }
+
+    /**
+     * @test
+     */
+    public function it_can_push_a_message_with_a_delay(): void
+    {
+        /** @var ClockInterface|MockInterface $clock */
+        $clock = $this->createMock(ClockInterface::class);
+
+        /** @psalm-suppress UndefinedInterfaceMethod */
+        $clock
```

Comment:
That methods returns an int, which is only set once while setting up the mock.

I'm going to replace it with the following to try to avoid confusion:
```php
->willReturn(1234567890 * ClockInterface::NANOS_PER_SECOND);
```